tensorboard/components/tf_tensorboard/tf-tensorboard.html

```diff
@@ -245,62 +230,69 @@
         },
         disabledTabs: String,
       },
+      observers: [
+        '_ensureCurrentDashboardStamped(_dashboardContainersStamped, mode)',
+      ],
       _isTabEnabled: function(tab) {
         if (this.disabledTabs != null &&
             this.disabledTabs.split(',').indexOf(tab) >= 0) {
           return false;
         }
         return true;
       },
-      _getModeFromIndex: function(modeIndex) {
-        var mode = this.tabs[modeIndex];
-        setString(TAB, mode, /*useLocalStorage=*/false);
-        return mode;
-      },
-      _isReloadDisabled: function(mode) {
-        return !this._debuggerDataEnabled && this._modeIsGraphs(mode);
-      },
-      _modeIsScalars: function(mode) {
-        return mode === "scalars";
-      },
-      _modeIsImages: function(mode) {
-        return mode === "images";
-      },
-      _modeIsAudio: function(mode) {
-        return mode === "audio";
+      _getModeFromIndex(tabs, modeIndex) {
+        return tabs[modeIndex];
       },
-      _modeIsGraphs: function(mode) {
-        return mode === "graphs";
+      _getDisplayForMode(mode, dashboard) {
+        return mode === dashboard ? 'block' : 'none';
       },
-      _modeIsEmbeddings: function(mode) {
-        return mode === "embeddings";
-      },
-      _modeIsDistributions: function(mode) {
-        return mode === "distributions";
+      _updateMode(mode) {
+        setString(TAB, mode, /*useLocalStorage=*/false);
       },
-      _modeIsHistograms: function(mode) {
-        return mode === "histograms";
+      _ensureCurrentDashboardStamped(containersStamped, mode) {
+        if (!containersStamped) {
+          return;
+        }
+        const container = this.$$(`.dashboard-container[data-mode=${mode}]`);
```

Comment:
Yep: `this.$$ ≅ function(selector) { return Polymer.dom(this).querySelector(selector); }`, I believe.

(And `[k=v]` is the [CSS attribute selector][1], if that's what you meant.)

[1]: https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors